packages/slice-machine/server/src/api/io/CustomType.ts

```diff
@@ -14,8 +15,21 @@ export function writeCustomType(path: string, customType: CustomTypeSM) {
   Files.write(path, CustomTypes.fromSM(customType));
 }
 
-export function renameCustomType(path: string, newCustomTypeName: string) {
-  const customType = readCustomType(path);
+export function writeCustomTypeTypes(path: string, customType: CustomTypeSM) {
+  const content = generateTypes({
+    customTypeModels: [CustomTypes.fromSM(customType)],
+  });
+
+  Files.write(path, content);
+}
+
+export function renameCustomType(
+  modelPath: string,
+  typesPath: string,
+  newCustomTypeName: string
+) {
+  const customType = readCustomType(modelPath);
   customType.label = newCustomTypeName;
-  writeCustomType(path, customType);
+  writeCustomType(modelPath, customType);
+  writeCustomTypeTypes(typesPath, customType);
```

Comment:
nothing changes on the types, why do we need to re-write them ?

Reply:
TSDocs on the generated type make use of the Slice Display name.